metadoc-js/src/main/scala/monaco/Monaco.scala

```diff
@@ -2624,3 +2680,19 @@ package worker {
 object Monaco extends js.Object {
   type MarkedString = String | js.Any
 }
+
+package common {
+
+  @ScalaJSDefined
+  trait IReference[T] extends IDisposable {
+    def `object`: T
+  }
+
+  object IReference {
```

Comment:
`IReference` signifies "interface" so I suggest we do something like this instead to also signal the lifecycle guarantees:

```scala
@ScalaJSDefined
class ImmortalReference[T](override val `object`: T) extends IReference[T] {
  override def dispose(): Unit = ()
}
```